plugins/coverbrowser.koplugin/main.lua

```diff
@@ -126,47 +131,186 @@ function CoverBrowser:addToMainMenu(menu_items)
     end
     sub_item_table[#modes].separator = true
     table.insert(sub_item_table, {
-        text = _("Use this mode everywhere"),
+        text = _("Hide book covers"),
+        enabled_func = function()
+            return filemanager_display_mode == "mosaic_image"
+                or filemanager_display_mode == "mosaic_text"
+                or filemanager_display_mode == "list_image_meta"
+                or filemanager_display_mode == "list_only_meta"
+                or filemanager_display_mode == "list_image_filename"
+        end,
         checked_func = function()
-            return BookInfoManager:getSetting("unified_display_mode")
+            return filemanager_display_mode == "mosaic_text" or filemanager_display_mode == "list_only_meta"
         end,
         callback = function()
-            if BookInfoManager:toggleSetting("unified_display_mode") then
-                self:setupHistoryDisplayMode(filemanager_display_mode)
-                self:setupCollectionDisplayMode(filemanager_display_mode)
-            end
+            if filemanager_display_mode == "mosaic_image" then
+                self:setupFileManagerDisplayMode("mosaic_text")
+            elseif filemanager_display_mode == "mosaic_text" then
+                self:setupFileManagerDisplayMode("mosaic_image")
+            elseif filemanager_display_mode == "list_only_meta" then
+                self:setupFileManagerDisplayMode("list_image_meta")
+            else
+                self:setupFileManagerDisplayMode("list_only_meta")
+            end  
         end,
     })
     table.insert(sub_item_table, {
-        text = _("History display mode"),
+        text = _("Replace book title with file name"),
         enabled_func = function()
-            return not BookInfoManager:getSetting("unified_display_mode")
+            return filemanager_display_mode == "list_image_meta"
+                or filemanager_display_mode == "list_image_filename"
+                or filemanager_display_mode == "list_only_meta"
```

Comment:
Maybe you can use something like
https://github.com/koreader/koreader/blob/526a1fb7276468137045d6227c8f62b58c75daa3/plugins/coverbrowser.koplugin/main.lua#L513
to shorten this and similar checks.